common-lib/src/main/scala/com/gu/mediaservice/lib/imaging/ImageOperations.scala

```diff
@@ -133,10 +133,33 @@ class ImageOperations(playPath: String) {
       transformSource = addImage(sourceFile)
       addOutput    = addDestImage(transformSource)(outputFile)
       _           <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains("image/tiff"))
+      _           <- checkForOutputFileChange(outputFile)
+    } yield outputFile
+  }
+
+  private def checkForOutputFileChange(f: File): Future[Unit] = Future {
+    val fileBits = f.getAbsolutePath.split("\\.").toList
+    val mainPart = fileBits.dropRight(1).mkString(".")
+    val extension = fileBits.last
+
+    // f2 is the blah-0 name that gets created from a layered tiff.
+    val f2 = new File(List(s"${mainPart}-0", extension).mkString("."))
+    if (f2.exists()) {
+      // f HAS been renamed to blah-0.  Rename it right back!
```

Comment:
Do we know why/where/how this happens? It's not clear to me from the detail in this PR.

Reply:
I will improve the comment.